sickbeard/webserve.py

```diff
@@ -1365,6 +1367,167 @@ def titler(x):
             action="displayShow"
         )
 
+    def manualSnatchSelect(self, show=None, season=None, episode=None, url=None, downCurQuality=None, quality=None, release_group=None, provider=None, name=None):
+        try:
+            show = int(show)  # fails if show id ends in a period SickRage/sickrage-issues#65
+            showObj = Show.find(sickbeard.showList, show)
+        except (ValueError, TypeError):
+            return self._genericMessage("Error", "Invalid show ID: %s" % str(show))
+
+        if showObj is None:
+            return self._genericMessage("Error", "Show not in show list")
+
+        if url is None or quality is None or release_group is None or provider is None or name is None:
+            return self._genericMessage("Error", "URL not valid")
+
+        # retrieve the episode object and fail if we can't get one
+        ep_obj = self._getEpisode(show, season, episode)
+        if isinstance(ep_obj, str):
+            return json.dumps({'result': 'failure'})
+
+        # make a queue item for it and put it on the queue
+        ep_queue_item = search_queue.ManualSelectQueueItem(ep_obj.show, ep_obj, season, episode, url, quality, release_group, provider, name)
+
+        sickbeard.searchQueueScheduler.action.add_item(ep_queue_item)
+
+        if not ep_queue_item.started and ep_queue_item.success is None:
+            return json.dumps(
+                {'result': 'success'})  # I Actually want to call it queued, because the search hasnt been started yet!
+        if ep_queue_item.started and ep_queue_item.success is None:
+            return json.dumps({'result': 'success'})
+        else:
+            return json.dumps({'result': 'failure'})
+
+
+    def manualSelect(self, show=None, season=None, episode=None):
+        # todo: add more comprehensive show validation
+        try:
+            show = int(show)  # fails if show id ends in a period SickRage/sickrage-issues#65
+            showObj = Show.find(sickbeard.showList, show)
+        except (ValueError, TypeError):
+            return self._genericMessage("Error", "Invalid show ID: %s" % str(show))
+
+        if showObj is None:
+            return self._genericMessage("Error", "Show not in show list")
+
+        main_db_con = db.DBConnection('cache.db')
+        sql_results = sql_return = {}
+
+        providers = [x for x in sickbeard.providers.sortedProviderList(sickbeard.RANDOMIZE_PROVIDERS) if x.is_active() and x.enable_backlog]
+        for curProvider in providers:
+            sql_return = main_db_con.select(
+            "SELECT * FROM " + curProvider.name + " WHERE episodes = ? AND season = ? AND indexerid = ?",
+            ["|" + episode + "|", season, show]
+            )
+            if sql_return:
+                sql_results.update({curProvider.name: sql_return})
+
+        if not sql_return:
+            return self._genericMessage("Error", "No release in cache")
+
```

Comment:
This is crappy, that I know for sure :) It is supposed to procuce an array like (I'm terrible in Python, so this is what it would look like in PHP or Java)
sql_results['provider1'] = results array
sql_results['provider2'] = results array

Then if we don't have any results, we throw an error. I think you're right and it shoud be `if not sql_results` but I don't think it will make any difference.